### PR TITLE
add checking for exact number of queries

### LINKED ISSUE #56: assert_number_of_queries support
I would like to be able to do

```ruby
test 'my operation performs only a single database query' do
  assert_number_of_queries(1) do
     get :index
  end
end
```

My particular use case is a scope that had `includes` which caused 2 additional queries where I didn't need. This is for testing the opposite - unnecessary loading of relations.

### PR DESCRIPTION
**Pull request by akostadinov**
fixes #56

Not doing the whole checklist to see if feature is accepted as it is now.

## Checklist

- [x] I've added tests for this change
- [ ] I've added a Changelog entry
- [ ] I've updated a documentation

### CODE REVIEW

**Comment by palkan** [comment]
I think, this one is very similar to constant queries matcher. Maybe we can extend it like this instead?

```rb
expect { subject }.to perform_constant_number_of_queries(exactly: 1)
```

**Comment by akostadinov** [comment]
There is some issue that I'll fix somehow, when I'm able to reproduce locally. But could you tell me if you like how code looks now?

I merged the rspec matchers. But left the minitest assertion because it doesn't seem to reuse much code. Is this fine with you or you want also minitest to use a parameter instead of a new assertion?

File: spec/n_plus_one_control/rspec/matchers/perform_exact_number_of_queries_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      end.to raise_error(RSpec::Expectations::ExpectationNotMetError)
+    end
+  end
+
+  context "exact number has precedence over scale factor", :n_plus_one do
+    specify do
+      expected_params = {population: nil, matching: /INSERT/, scale_factors: [1]}
+      executor = NPlusOneControl::Executor.new(**expected_params)
+      expect(NPlusOneControl::Executor).to receive(:new).with(**expected_params).and_return(executor)
+      expect { Post.take }.to perform_constant_number_of_queries.exactly(0).matching(/INSERT/).with_scale_factors(2, 3)
```
Comment (akostadinov) [inline spec/n_plus_one_control/rspec/matchers/perform_exact_number_of_queries_spec.rb:31]: For some reason if I use `exactly(1)` with the default `SELECT` matching, in rails 7 tests **two** queries are more often than not detected. When I used pry to debug why, it didn't happen. So there might be some race in newer rails versions. That's why I decided to use `matching(/INSERT/)` as a workaround.

But it is still super strange. Do you have any possible explanation?

**Comment by palkan** [comment]
Finally merged and released (see 0.7.1, it also includes some other minor things we discussed a while ago).

I changed the final API for Minitest: similarly to RSpec, I extended the existing `#assert_perform_constant_number_of_queries` API to support passing the exact number: we just pass a positional argument

```ruby
assert_perform_constant_number_of_queries(2, **options) do
  get :index
end
```

**Pull request merged**